src/htmx.js

```diff
@@ -1141,31 +1141,51 @@ var htmx = (function() {
     elt = resolveTarget(elt)
-    if (selector.indexOf('closest ') === 0) {
-      return [closest(asElement(elt), normalizeSelector(selector.substr(8)))]
-    } else if (selector.indexOf('find ') === 0) {
-      return [find(asParentNode(elt), normalizeSelector(selector.substr(5)))]
-    } else if (selector === 'next') {
-      return [asElement(elt).nextElementSibling]
-    } else if (selector.indexOf('next ') === 0) {
-      return [scanForwardQuery(elt, normalizeSelector(selector.substr(5)), !!global)]
-    } else if (selector === 'previous') {
-      return [asElement(elt).previousElementSibling]
-    } else if (selector.indexOf('previous ') === 0) {
-      return [scanBackwardsQuery(elt, normalizeSelector(selector.substr(9)), !!global)]
-    } else if (selector === 'document') {
-      return [document]
-    } else if (selector === 'window') {
-      return [window]
-    } else if (selector === 'body') {
-      return [document.body]
-    } else if (selector === 'root') {
-      return [getRootNode(elt, !!global)]
-    } else if (selector.indexOf('global ') === 0) {
-      return querySelectorAllExt(elt, selector.slice(7), true)
-    } else {
-      return toArray(asParentNode(getRootNode(elt, !!global)).querySelectorAll(normalizeSelector(selector)))
+
+    const parts = selector.split(',')
+    const result = []
+    const unprocessedParts = []
+    for (let i = 0; i < parts.length; i++) {
+      let selector = parts[i].trimStart()
+
+      if (selector.indexOf('global ') === 0) {
+        global = true
+        selector = selector.substring(7)
+        parts[i] = selector
+      }
+
+      if (selector.indexOf('closest ') === 0) {
+        result.push(closest(asElement(elt), normalizeSelector(selector.substr(8))))
+      } else if (selector.indexOf('find ') === 0) {
+        result.push(find(asParentNode(elt), normalizeSelector(selector.substr(5))))
+      } else if (selector === 'next' || selector === 'nextElementSibling') {
+        result.push(asElement(elt).nextElementSibling)
+      } else if (selector.indexOf('next ') === 0) {
+        result.push(scanForwardQuery(elt, normalizeSelector(selector.substr(5)), !!global))
+      } else if (selector === 'previous' || selector === 'previousElementSibling') {
+        result.push(asElement(elt).previousElementSibling)
+      } else if (selector.indexOf('previous ') === 0) {
+        result.push(scanBackwardsQuery(elt, normalizeSelector(selector.substr(9)), !!global))
+      } else if (selector === 'document') {
+        result.push(document)
+      } else if (selector === 'window') {
+        result.push(window)
+      } else if (selector === 'body') {
+        result.push(document.body)
+      } else if (selector === 'root') {
+        result.push(getRootNode(elt, !!global))
+      } else {
+        // Push back the unaltered string part, so that the standard selector fallback uses the exact initial string
+        unprocessedParts.push(parts[i])
```

Comment:
push selector here and you won't have to reassign parts[i] = selector if global

Reply:
not sure if trimming whitespace would cause issues though??  my testing in chrome shows whitespace ignored but is this the same for all browsers?

Reply:
Just noticed that the `querySelectorAll` call below is using `normalizeSelector` anyway so... you're right, this can be simplified, whitespaces were always being removed it seems!